students/chrissp/lesson04/mailroom.py

```diff
@@ -46,17 +51,20 @@ def send_thanks():
             main()
         else:
             for person in donor_db:
-                if donor_name == person[0]:
-                    this_donor = person[0]
-                    person[1].append(int(this_donation))
+                if user_input == person["name"]:
+                    this_donor = person
+                    person["donations"].append(int(this_donation))
                     break
 
             if this_donor == "":
-                donor_db.append((donor_name, [int(this_donation)]))
+                donor_db.append({"name": user_input, "donations": [int(this_donation)]})
+
+            this_letter = {'donor_name': user_input,
+                           'last_donation': this_donation,
+                           'total_donations': len(this_donor["donations"])}
 
             # Build thank you email
-            print(f"\n\nDear {donor_name},\n Thank you for your generous donation of ${this_donation}!\n",
-                  "Sincerly,\n Scott's Tots Foundation\n\n")
+            print(letter_template.format(**this_letter))
```

Comment:
Getting an error when adding a new donor and donation for this function.  Can you take another look?  Let me know if you have questions.